docs/guides/color.md

```diff
@@ -0,0 +1,290 @@
+
+The [`transform_color`](https://github.com/napari/napari/blob/5f96d5d814aad697c367bdadbb1a57750e2114ad/napari/utils/colormaps/standardize_color.py#L33)
+function is responsible for coercing color input values to the standard form and supports a variety of inputs.
+
+```{note}
+The [`ColorType`](https://github.com/napari/napari/blob/43d9b59da89b9a2b97dba3967a7b28526f9aa0ce/napari/utils/colormaps/colormap_utils.py#L17)
+type definition almost describes all the types of acceptable color inputs, except that it includes
+vispy's `Color` and `ColorArray`, which are not accepted by `transform_color`.
+```
+
+
+### Supported single color input
+
+The single face color could have been defined as red in many equivalent ways.
+
+```python
+# CSS3 name
+points.face_color = 'red'
+
+# matplotlib single character name
+points.face_color = 'r'
+
+# RGB or RGBA hex-code
+points.face_color = '#ff0000'
+points.face_color = '#ff0000ff'
+
+# RGB or RGBA tuple, list, or array
+points.face_color = (1, 0, 0)
+points.face_color = (1, 0, 0, 1)
+points.face_color = [1, 0, 0]
+points.face_color = [1, 0, 0, 1]
+points.face_color = np.array([1, 0, 0])
+points.face_color = np.array([1, 0, 0, 1])
+```
+
+A named color string must either be from the [CSS3 specification](https://www.w3.org/TR/css-color-3/#svg-color)
+or must be one of [matplotlib's single character shorthand names](https://matplotlib.org/stable/tutorials/colors/colors.html#specifying-colors).
+
+
+### Supported multiple color input
+
+Sometimes there is a need to specify many colors at once.
+For example, the face colors of all the points in a layer can be specified individually
+
+```python
+In [3]: points.face_color = ['red', 'lime', 'blue']
+```
+
+which will then be coerced into the standard array form.
+
+```python
+In [4]: points.face_color
+Out[4]:
+array([[1., 0., 0., 1.],
+       [0., 1., 0., 1.],
+       [0., 0., 1., 1.]])
+```
+
+For the most part, any tuple, list, array, or generator of single colors
+is acceptable input when specifying multiple colors.
+
+```python
+# Tuple, list, array, or generator of names
+points.face_color = ('red', 'lime', 'blue')
+points.face_color = ['red', 'lime', 'blue']
+points.face_color = np.array(['red', 'lime', 'blue'])
+points.face_color = c for c in ('red', 'lime', 'blue')
+
+# List of RGB or RGBA tuples or lists
+points.face_color = [
+    (1, 0, 0),
+    (0, 1, 0),
+    (0, 0, 1),
+]
+points.face_color = [
+    [1, 0, 0, 1],
+    [0, 1, 0, 1],
+    [0, 0, 1, 1],
+]
+
+# (N, 3) RGB or (N, 4) RGBA array
+points.face_color = np.array([
+    [1, 0, 0],
+    [0, 1, 0],
+    [0, 0, 1],
+])
+points.face_color = np.array([
+    [1, 0, 0, 1],
+    [0, 1, 0, 1],
+    [0, 0, 1, 1],
+])
+```
+
+However, when specifying many colors in a sequence or generator *do not mix different single color types*.
+For example, something like
+
+```python
+In [5]: points.face_color = np.array(['red', (0, 1, 0), '#0000ff'])
+```
+
+is not supported or guaranteed to work.
+
+
+## Colormaps
```

Comment:
I think one of the major reasons people search for colormaps in the napari documentation is so they can see the list of available colormaps. Do we have a list available? I understand that might work best as a supporting page to this one, since this page is explanation, and the reference list would be, well, a reference.

From memory, I think it includes most (but not all) matplotlib colormaps, plus also some extra colormaps other people made. Most napari colormaps are visible from the colormap drowdown box in the napari viewer, but not all of them are.

I'm always visiting the [matplotlib colormap page](https://matplotlib.org/stable/tutorials/colors/colormaps.html) to see a list with images beside them.

Reply:
I agree it is nice to have a visual representation of available colormaps.

The `Image` layer tutorial has a [section on available colormaps](
https://napari.org/tutorials/fundamentals/image.html#working-with-colormaps). It gives a function to see the available colormaps, but doesn't show them. I wonder if we could add a page that autogenerates colormap swatches from this function.

Reply:
Yep, like @kevinyamauchi mentioned the user can call `list(napari.utils.colormaps.AVAILABLE_COLORMAPS)`, which is also mentioned in this doc.

Note that `AVAILABLE_COLORMAPS` is mutable, though is initially populated with the colormaps you suspected (also described in this doc). In particular, you can use a colormap that isn't present initially using a uniquely vispy name (e.g. `image.colormap = 'coolwarm'`), which is not initially in `AVAILABLE_COLORMAPS`, but will be added.

My guess is the default colormaps will not change too frequently, so it would probably be OK to generate a one-off plot of the current ones and include that as an image. But ideally, they would be autogenerated right here, maybe with some surrounding text describing each set of colormaps (as the matplotlib docs do). Should that be fairly straightforward @GenevieveBuckley ?

Reply:
Maybe this is the occation to take a page out of [vispy's "autogenerated gallery"'s book](https://github.com/vispy/vispy/blob/main/vispy/util/gallery_scraper.py)?

Reply:
I don't think exhaustivity is necessary here, I would go for a curated list of good colormaps.
Otherwise you will start to expose things like `Jet`,  or `Spectral` vs `spectral`

Reply:
@Carreau : I didn't realize that our colormap names are case sensitive.

There seems to some split opinion on whether this should be exhaustive or not. I lean towards including an exhaustive reference of all the default colormaps for the reasons mentioned by @GenevieveBuckley @kevinyamauchi. So maybe let me try to generate that and see what we think?

Reply:
I think the case sensitivity is a (matplotlib) historical artifact that they are trying to get rid of. Maybe they even already did in latests, but it used to be the case.

Reply:
In the UI we only expose a curated list for the reasons mentioned by @Carreau, and I would suggest that we do the same here: expose a curated list, then link out to matplotlib's docs to say, "any matplotlib colormap will also work if provided as a name with the API". That is actually better (imho) because it ensures that we stay up to date if matplotlib updates, since that is technically what we are doing: calling out to matplotlib when the colormap is not in our prepopulated list.

Reply:
Oops, I didn't actually realize that we support every matplotlib colormap (at least the ones that I tried), even the qualitative ones. In that case, agreed that we should have a curated list and a link out to the matplotlib docs.

I think the initial/default value of `AVAILABLE_COLORMAPS` is the curated list we want - it's what we initially see in the GUI and there's only about 20 colormaps.

Reply:
I agree the default `AVAILABLE_COLORMAPS` is probably the right selection of colormaps to show. If it turns out to be a ton of effort, I am open to adding it in a follow-up PR, as this PR is already a solid chunk of work and is super informative as-is!